L1Trigger/TrackFindingTMTT/src/Histos.cc

```diff
@@ -0,0 +1,3971 @@
+            // Only prints for tracks with huge number of stubs, to also understand why these tracks exist.
+            //if (trk.getNumStubs() > 20) {
+            /*
+	    if (trk.pt() > 20) { 
+	    cout<<"--- Checking how far stubs on matched tracks lie from true particle trajectory. ---"<<endl;
+	    cout<<"    Track "<<trk.getPurity()<<" "<<tp->pt()<<" "<<tp->d0()<<endl;
+	    float sigPhiR = deltaPhiR/s->sigmaPerp();
+	    float sigRorZ = deltaRorZ/s->sigmaPar();
+	    string ohoh =  (fabs(sigPhiR) > 5 || fabs(sigRorZ) > 5)  ?  "FAR"  :  "NEAR";
+	    if (trueStub) {
+	    cout<<"    Real stub "<<ohoh<<" ps="<<s->psModule()<<" bar="<<s->barrel()<<" lay="<<s->layerId()<<" : phi="<<deltaPhiR<<" ("<<sigPhiR<<") rz="<<deltaRorZ<<" ("<<sigRorZ<<")"<<endl;
+	    } else {
+	    cout<<"    FAKE stub "<<ohoh<<" ps="<<s->psModule()<<" bar="<<s->barrel()<<" lay="<<s->layerId()<<" : phi="<<deltaPhiR<<" ("<<sigPhiR<<") rz="<<deltaRorZ<<" ("<<sigRorZ<<")"<<endl; 
+	    }
+	    cout<<"        coords="<<s->r()<<" "<<s->phi()<<" "<<s->eta()<<" bend="<<s->bend()<<" iphi="<<s->iphi()<<endl;
+	    cout<<"        module="<<s->minR()<<" "<<s->minPhi()<<" "<<s->minZ()<<endl;
+	    }
+	  */
+          }
+        }
+      }
+    }
+
+    // Count total number of tracking particles in the event that were reconstructed,
+    // counting also how many of them were reconstructed multiple times (duplicate tracks).
+
+    unsigned int nRecoedTPsForEff =
+        0;  // Total no. of TPs used for the efficiency measurement that were reconstructed as at least one track.
+    unsigned int nRecoedTPs = 0;           // Total no. of TPs that were reconstructed as at least one track.
+    unsigned int nEtaSecsMatchingTPs = 0;  // Total no. of eta sectors that all TPs were reconstructed in
+    unsigned int nSecsMatchingTPs = 0;     // Total no. of eta x phi sectors that all TPs were reconstructed in
+    unsigned int nTrksMatchingTPs = 0;     // Total no. of tracks that all TPs were reconstructed as
+
+    for (const TP& tp : vTPs) {
+      vector<const L1track3D*> matchedTrks;
+      for (const L1track3D& trk : tracks) {
+        const TP* tpAssoc = trk.getMatchedTP();
+        if (tpAssoc != nullptr) {
+          if (tpAssoc->index() == tp.index())
+            matchedTrks.push_back(&trk);
+        }
+      }
+      unsigned int nTrk = matchedTrks.size();
+
+      bool tpRecoed = false;
+
+      if (nTrk > 0) {
+        tpRecoed = true;           // This TP was reconstructed at least once in tracker.
+        nTrksMatchingTPs += nTrk;  // Increment sum by no. of tracks this TP was reconstructed as
+
+        set<unsigned int> iEtaRegRecoed;
+        for (const L1track3D* trk : matchedTrks)
+          iEtaRegRecoed.insert(trk->iEtaReg());
+        nEtaSecsMatchingTPs = iEtaRegRecoed.size();
+
+        set<pair<unsigned int, unsigned int>> iSecRecoed;
+        for (const L1track3D* trk : matchedTrks)
```

Comment:
Only a dummy version of class Histos, with no implementation, will be included in the updated PR. (This will allow us to replace it will the full implementation for single-threaded L1 track development work, but will avoid the problem of histograms preventing multithreaded EDProducers for CMSSW official code)